pkg/operator/bootstrap.go

```diff
@@ -140,7 +140,7 @@ func (c *Controller) Bootstrap() error {
 			return err
 		}
 	} else {
-		err = driver.CompleteConfiguration()
+		err = driver.CompleteConfiguration(&cr.Status)
```

Comment:
how do we get here if initially a registry resource already exists, with no storage configuration, and then someone updates the registry resource to include storage configuration?  (as used to be the case before you added the s3 auto-configuration, and is still the case for other storage types today)

does that even work anymore?  (supplying the storage configuration after the fact)

Reply:
afaik we would not reach the driver.CompleteConfiguration() function, it is only called from this one spot in bootstrap.go.

Reply:
My idea was that the `CompleteConfiguration` function should be called only to complete the configuration from global config when creating a new custom resource. Since the storage type cannot be changed, it is not necessary to coplete the configuration in other cases. A configuration change will cause an error.

> "no storage configured" to "use this storage configuration" should send us through the driver configuration codepath.

@bparees `no storage configured`  can _only_ happen if someone creates custom resource (without storage section) before operator first start. If the operator has already been started, then it will add the finalizer and you will not be able to delete the resource or change the storage type.

Reply:
Well those assumptions are no longer valid since "CompleteConfiguration" is now being used to actually create the buckets, not just fill out the storage configuration spec.

Either the bucket creation needs to happen elsewhere, or CompleteConfiguration needs to be invoked when the user fills in the storage type later.

Reply:
@bparees in this case we need to refactor our clusterconfig. It will [panic if S3 will be used on a non-AWS platform](https://github.com/openshift/cluster-image-registry-operator/blob/add05f58a0dc895f76afae44147008b287e420f1/pkg/clusterconfig/clusterconfig.go#L103).

Reply:
yes we will have to revisit that at some point.

Reply:
@legionus or we revisit where the create bucket is being done, I don't care what approach is taken.  I am trying to raise your awareness to scenarios that you need to consider and handle properly.  I do not care how you address them (though of course i may have feedback when i see the proposed change).

Reply:
> I do not care how you address them (though of course i may have feedback when i see the proposed change).

@bparees as you wish.